llm_service/main.py

```
@@ -0,0 +1,78 @@
+                print("[DEBUG] Inference complete")
+                return embeddings[0].tolist()
+        except Exception as e:
+            print("[ERROR] Exception during inference:", e)
+            raise
+
+    loop = asyncio.get_event_loop()
+    try:
+        vector = await loop.run_in_executor(None, run_inference)
+    except Exception as e:
+        print("[ERROR] Inference failed:", e)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)